packages/amplication-client/src/VersionControl/Commit.tsx

```diff
@@ -88,59 +136,182 @@ const Commit = ({
-                  className={`${CLASS_NAME}__technology-toggle`}
-                  label=""
-                  name="action_"
-                  options={OPTIONS}
-                  onChange={handleOnSelectLanguageChange}
-                  selectedValue={"node"}
-                />
-              )}
-              <CommitButton
-                commitBtnType={commitBtnType}
-                commitMessage={formikRef.current?.values?.message}
-                onCommitChanges={handleCommitBtnClicked}
-              ></CommitButton>
-            </Form>
-          );
-        }}
-      </Formik>
-    </div>
+        <FlexItem
+          direction={EnumFlexDirection.Column}
+          itemsAlign={EnumItemsAlign.Start}
+        >
+          {resources
+            .filter((r) => r.resourceType === EnumResourceType.Service)
+            .map((resource, index) => (
+              <Button
+                buttonStyle={EnumButtonStyle.Text}
+                onClick={() => {
+                  handleOnSpecificServiceCommit(resource.id);
+                }}
+              >
+                <FlexItem direction={EnumFlexDirection.Row}>
+                  <ResourceCircleBadge
+                    type={EnumResourceType.Service}
+                    size="small"
+                  />
+                  {resource.name}
+                </FlexItem>
+              </Button>
+            ))}
+        </FlexItem>
+      </Dialog>
+      <div className={CLASS_NAME}>
+        <Formik
+          initialValues={INITIAL_VALUES}
+          onSubmit={handleSubmit}
+          validateOnMount
+          innerRef={formikRef}
+        >
+          {(formik) => {
+            const handlers = {
+              SUBMIT: formik.submitForm,
+            };
+
+            return (
+              <Form>
+                {!commitChangesLoading && (
+                  <GlobalHotKeys keyMap={keyMap} handlers={handlers} />
+                )}
+                {showCommitMessage && (
+                  <TextField
+                    rows={3}
+                    textarea
+                    name="message"
+                    label={noChanges ? "Build message" : "Commit message..."}
+                    disabled={commitChangesLoading}
+                    autoFocus
+                    hideLabel
+                    placeholder={
+                      noChanges ? "Build message" : "Commit message..."
+                    }
+                    autoComplete="off"
+                  />
+                )}
+                {!dotNetGeneratorEnabled && (
+                  <MultiStateToggle
+                    className={`${CLASS_NAME}__technology-toggle`}
+                    label=""
+                    name="action_"
+                    options={OPTIONS}
+                    onChange={handleOnSelectLanguageChange}
+                    selectedValue={"node"}
+                  />
+                )}
+                {commitBtnType === CommitBtnType.Button ? (
```

Comment:
If I understand correctly, you are using the commitBtnType to "understand" in which page you are in right now? and then show or hide the selectMenu options?
If that is the case, let's at least create a small utility method that says if we need to add the SelectMenu or not - inside the function it can still use the commitBtnType to decide but at least it will be encapsulated - when it appears here in this way it does look a bit like a patch or some "code smell"

Reply:
@mulygottlieb Condition removed (Irrelevant anymore).